spec/features/course_page_spec.rb

```diff
@@ -426,4 +426,61 @@ def js_visit(path, count=3)
       end
     end
   end
+
+  describe 'articles tracking' do
+    let(:course) do
+      editathon_slug = 'This_university.foo.bar/This.course_(term_2015)'
+      create(:editathon,
+             id: 10002,
+             title: 'This.course',
+             slug: editathon_slug,
+             start: '10-07-2019'.to_date,
+             end: '10-08-2019'.to_date,
+             timeline_start: course_start.to_date,
+             timeline_end: course_end.to_date,
+             school: 'This university.foo.bar',
+             expected_students: 1,
+             term: 'term 2015',
+             home_wiki_id: home_wiki.id,
+             description: 'This is a great course',
+             flags: update_logs)
+    end
+    let(:user) { create(:user, username: 'Hbultra') }
+
+    before do
+      course.students << user
+      VCR.use_cassette 'course_upload_importer/Hbultra' do
```

Comment:
Because this spec is aimed at the frontend rather than the tracked data model itself, doing this network-reliant data update during the spec is not ideal. It will be slower and more fragile than it needs to be, and it doesn't test anything related to the data updates that isn't tested elsewhere already. I'd rather see a tracked article added to the course explicitly in order to set up the right situation where the filters and toggles appear.